File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        else:
            sum=0
            temp = x
            while x > 0:
                r = x % 10
                sum = sum *10 + r
                x = x//10
            print(sum)
            if int(sum) == temp:
                return True
            else:
                return False
        

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
            if(x<0){
                return false;
            }
            else{
                int total=0;
                int dup = x;
                while(x>0){
                    int r = x%10;
                    if((total < INT_MIN/10) || (total > INT_MAX/10)){
                        return false;
                    }
                    total = total * 10 + r;
                    x = x/10;
                }
                if(dup==total){
                    return true;
                }
                else{
                    return false;
                }
                
            }
        }
};